### PR TITLE
Fixes bi-directional bucket replication tests issue

### LINKED ISSUE #9131: Bidirectional bucket replication tests require different prefixes between the source and the destination
Due to the recent changes that were added in https://github.com/noobaa/noobaa-core/pull/7524, bidirectional bucket replication is no longer supported when the two policies use the same prefix for both the source and the destination. That includes when the two policies are using an empty prefix - as is the case in our OCS-CI tests.

Additional reference: https://url.corp.redhat.com/4b2558b

### PR DESCRIPTION
**Pull request by udaysk23**
Fixes #9131

### CODE REVIEW

File: tests/functional/object/mcg/test_bucket_replication.py [hunk 1]
```diff
@@ -276,32 +276,41 @@ def test_bidirectional_bucket_replication(
...
+            test_directory_setup.origin_dir,
+            amount=10,
+            prefix=prefix_site1,
+            mcg_obj=mcg_obj_session,
         )
-        first_bucket_set = set(standard_test_obj_list)
-        assert first_bucket_set == {
+        written_objects_with_prefix = list(
+            map(lambda x: prefix_site1 + "/" + x, written_objects)
+        )
```
Comment (sagihirshfeld) [inline tests/functional/object/mcg/test_bucket_replication.py:308]: The use of `map` is clever but a bit unreadable IMO. How about this for simplicity?
```suggestion
        written_objects_with_prefix = [f"{prefix_site1}/{obj}" for obj in written_objects]
```

File: tests/functional/object/mcg/test_bucket_replication.py [hunk 2]
```diff
@@ -310,10 +319,14 @@ def test_bidirectional_bucket_replication(
...
             test_directory_setup.origin_dir,
-            amount=5,
+            amount=10,
+            prefix=prefix_site2,
             mcg_obj=mcg_obj_session,
         )
-        second_bucket_set = set(written_objects)
+        written_objects_with_prefix = list(
+            map(lambda x: prefix_site2 + "/" + x, written_objects)
+        )
```
Comment (sagihirshfeld) [inline tests/functional/object/mcg/test_bucket_replication.py:328]:
```suggestion
        written_objects_with_prefix = [f"{prefix_site2}/{obj}" for obj in written_objects]
```

**Comment by sagihirshfeld** [comment]
Also please set the `Verified` label since the PR passed the verification - it helps bring attention for reviews and merging.
Other than that and the two semantic suggestions above - LGTM 👍 

Once you decide to reject or accept the suggestions I'll approve the PR

File: tests/functional/object/mcg/test_bucket_replication.py (hunk 1 above)
Comment (udaysk23) [reply tests/functional/object/mcg/test_bucket_replication.py]: Yes we can use both logic. I will go ahead and add for loop for simplicity

**Review by sagihirshfeld — APPROVED** [review]

**Review by tiffanyn108 — APPROVED** [review]

**Pull request merged**